kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/TestUtils.scala

```diff
@@ -79,6 +80,7 @@ trait TestUtils extends Assertions {
   def readSnapshot(
     snapshot: Snapshot,
     readSchema: StructType = null,
+    filter: Predicate = null,
```

Comment:
why are we using `null` instead of Optional?

Reply:
IMO, following the existing patterns. Also it reduces the test verbose a bit. If this was production code, I would have used the Optional.